pyoverkiz/client.py

```diff
@@ -155,6 +173,78 @@ async def login(
 
         return False
 
+    async def somfy_tahoma_get_access_token(self) -> str:
+        """
+        Authenticate via Somfy identity and acquire access_token.
+        """
+        # Request access token
+        async with self.session.post(
+            SOMFY_API + "/oauth/oauth/v2/token",
+            data=FormData(
+                {
+                    "grant_type": "password",
+                    "username": self.username,
+                    "password": self.password,
+                    "client_id": SOMFY_CLIENT_ID,
+                    "client_secret": SOMFY_CLIENT_SECRET,
+                }
+            ),
+            headers={
+                "Content-Type": "application/x-www-form-urlencoded",
+            },
+        ) as response:
+            token = await response.json()
+            # { "message": "error.invalid.grant", "data": [], "uid": "xxx" }
+            if "message" in token and token["message"] == "error.invalid.grant":
+                raise SomfyBadCredentialsException(token["message"])
+
+            if "access_token" not in token:
+                raise SomfyServiceException("No Somfy access token provided.")
+
+            self._access_token = str(token["access_token"])
+            self._refresh_token = str(token["refresh_token"])
+            self._expires_in = datetime.datetime.now() + datetime.timedelta(
+                seconds=int(token["expires_in"]) - 5
+            )
+
+            return self._access_token
+
+    async def somfy_tahoma_refresh_token(self, refresh_token: str) -> str:
+        """
+        To get a new access token, the refresh token is needed. The refresh_token will be valid 14 days.
+        """
+        # &grant_type=refresh_token&refresh_token=REFRESH_TOKEN
+        # Request access token
+        async with self.session.post(
+            SOMFY_API + "/oauth/oauth/v2/token",
+            data=FormData(
+                {
+                    "grant_type": "refresh_token",
+                    "refresh_token": refresh_token,
+                    "client_id": SOMFY_CLIENT_ID,
+                    "client_secret": SOMFY_CLIENT_SECRET,
+                }
+            ),
+            headers={
+                "Content-Type": "application/x-www-form-urlencoded",
+            },
+        ) as response:
+            token = await response.json()
+            # { "message": "error.invalid.grant", "data": [], "uid": "xxx" }
+            if "message" in token and token["message"] == "error.invalid.grant":
+                raise SomfyBadCredentialsException(token["message"])
+
+            if "access_token" not in token:
+                raise SomfyServiceException("No Somfy access token provided.")
+
+            self._access_token = str(token["access_token"])
+            self._refresh_token = str(token["refresh_token"])
```

Comment:
They are not already a string?

Reply:
I think so... Would need to double check. Same for the int.